traitlets/traitlets.py

```diff
@@ -489,11 +488,28 @@ def _validate(self, obj, value):
         return value
 
     def _cross_validate(self, obj, value):
-        if hasattr(obj, '_%s_validate' % self.name):
-            cross_validate = getattr(obj, '_%s_validate' % self.name)
-            value = cross_validate(value, self)
+        try:
+            cb = getattr(obj, '_%s_validate' % self.name)
+        except AttributeError:
+            cb = obj._trait_validators.get(self.name,None)
+        else:
+            warn("_[traitname]_validate handlers are deprecated: use register_validator instead",
+                 DeprecationWarning, stacklevel=2)
+            if not callable(cb):
+                raise TraitError('A trait validator must be callable')
+            if obj._trait_validators.get(self.name,None):
+                raise TraitError('Only one cross-validator is allowed: two were found')
+            return cb(value, self)
+
+        if cb:
+            if callable(cb):
```

Comment:
Please just call it.  If it's not callable, an error will be propagated to the user automatically.